### ISSUE TITLE
iBoot64Patch: Failed getting sigpatches

### ISSUE THREAD

**Issue by JomSpoons** [marijuanARM/futurerestore#162]
Hello. I'm trying to restore my iPad (model A1822 & identifier 6,11) to iOS 16.6.1 using blobs and Legacy-iOS-Kit, but during the process I get an error that says `iBoot64Patch: Failed getting sigpatches!` I was told by LukeZGD that this is an issue with futurerestore in specific and better is suited for this repository. Here is the context for the error:

```
Verified APTicket to be valid for this restore
Variant: Customer Erase Install (IPSW)
This restore will erase all device data.
irecv_event_cb: device 00000575f232d1f9 (udid: N/A) connected in DFU mode
Device found in DFU Mode.
Getting firmware keys for: j71sap
Patching iBSS
Extracting iBSS.ipad6f.RELEASE.im4p (Firmware/dfu/iBSS.ipad6f.RELEASE.im4p)...
payload decrypted
iBoot64Patch: Staring iBoot64Patch!
iOS 16 iBoot detected!
iBoot64Patch: Inited ibootpatchfinder64!
iBoot64Patch: Failed getting sigpatches!
Cleaning up...
[exception]:
what=assure failed
code=26476559
line=404
file=libipatcher.cpp
commit count=91
commit sha  =cb10d973d0af78cc55020d4cf1187c28fad0f2a0
Done: restoring failed!
```

I've also attached the full log from Legacy-iOS-Toolkit if more info is needed. I'd greatly appreciate any help in resolving this issue. Thank you!
[iboot64patch error.txt](https://github.com/futurerestore/futurerestore/files/15351062/iboot64patch.error.txt)

**Comment by Cryptiiiic** [comment]
I'll look into it, hopefully I don't forget. Thanks for the report.

**Comment by Cryptiiiic** [comment]
@JomSpoons can you try the build linked here? https://github.com/futurerestore/futurerestore/actions/runs/9229546846

**Comment by JomSpoons** [comment]
> @JomSpoons can you try the build linked here? https://github.com/futurerestore/futurerestore/actions/runs/9229546846

Hey, sorry for the super late response. I had never used futurerestore independently from Legacy-iOS-Kit so I was confused on how to proceed and ended up putting it off for a while. I figured I should use [the most recent artifact](https://github.com/futurerestore/futurerestore/actions/runs/9456430852) rather than the one above linked, so that's what I ended up doing. I also used FutureRestore GUI to make things easier for me to understand, hopefully that doesn't have any impact on the results.

I got a lot further in the restoration process this time, however it somehow ended up putting me on the latest available iPadOS available for my device (16.7.8) rather than the one I selected the blob and IPSW for (16.6.1). Not really sure what went wrong but I've attached the full log. Regardless, thank you very much for your help thus far!

[FRLog_2024-06-26T20.08.58.587967.txt](https://github.com/user-attachments/files/16006992/FRLog_2024-06-26T20.08.58.587967.txt)
